internal/middleware/connlimit.go

```diff
@@ -0,0 +1,36 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/centrifugal/centrifugo/v4/internal/rule"
+
+	"github.com/centrifugal/centrifuge"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	connLimitReached prometheus.Counter
+)
+
+func init() {
+	connLimitReached = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "centrifugo",
+		Subsystem: "node",
+		Name:      "client_connection_limit",
+		Help:      "Number of refused requests due to node client connection limit.",
+	})
+	_ = prometheus.DefaultRegisterer.Register(connLimitReached)
+}
+
+func ConnLimit(node *centrifuge.Node, ruleContainer *rule.Container, h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		connLimit := ruleContainer.Config().ClientConnectionLimit
+		if connLimit > 0 && node.Hub().NumClients() >= connLimit {
+			connLimitReached.Inc()
+			node.Log(centrifuge.NewLogEntry(centrifuge.LogLevelWarn, "node connection limit reached", map[string]interface{}{"limit": connLimit}))
```

Comment:
Other minor comment: Since there is a metric that can be watched, I would make this an Info log, instead of a Warn. But that's likely preference/philosophy.

Reply:
I'd say the situation is quite abnormal, it requires an action from administrator - so warn seems fine here. 

@cristaloleg when will we have a performant logger library with possibility to tune a level of individual log lines if required and throttle the frequently repeated log lines, I think you mentioned it recently? 😃

Reply:
Added some throttling in b12fa7e13bc6c52bd202b4f5202eab866ddcfa40 to avoid too many warns in logs.